File: ChessProject-PHP/tests/PawnTest.php
<?php

namespace SolarWinds\Chess;

use SolarWinds\Chess\ChessBoard;
use SolarWinds\Chess\Pawn;
use SolarWinds\Chess\PieceColorEnum;

class PawnTest extends \PHPUnit_Framework_TestCase
{
    /** @var  ChessBoard */
    private $_chessBoard;
    
    /** @var  Pawn */
    private $_testSubject;
    
    /** @var  Pawn */
    private $_testBlack;
    
    protected function setUp () {
        $this->_chessBoard = new ChessBoard();
        $this->_testSubject = new Pawn(PieceColorEnum::WHITE());
        $this->_testBlack = new Pawn(PieceColorEnum::BLACK());
    }
    
    public function testChessBoard_Add_Sets_XCoordinate () {
        $this->_chessBoard->add($this->_testSubject, 6, 3);
        $this->assertEquals(6, $this->_testSubject->getXCoordinate());
    }
    public function testChessBoard_Add_Sets_YCoordinate () {
        $this->_chessBoard->add($this->_testSubject, 6, 3);
        $this->assertEquals(3, $this->_testSubject->getYCoordinate());
    }
    
    public function testPawn_Move_IllegalDestination_High () {
        $height = $this->_chessBoard->getSquareHeight();
        $this->_chessBoard->add($this->_testSubject, 0, $height-1);
        
        $this->expectException(\InvalidArgumentException::class);
        $this->_testSubject->move(0, $height);
    }
    public function testPawn_Move_IllegalDestination_Low () {
        $this->_chessBoard->add($this->_testBlack, 0, 0);
        
        $this->expectException(\InvalidArgumentException::class);
        $this->_testBlack->move(0, -1);
    }
    public function testPawn_Move_IllegalCoordinates_Right_DoesNotMove () {
        $this->_chessBoard->add($this->_testSubject, 6, 3);
        $this->expectException(\InvalidArgumentException::class);
        $this->_testSubject->move(7, 3);
    }
    public function testPawn_Move_IllegalCoordinates_Left_DoesNotMove () {
        $this->_chessBoard->add($this->_testSubject, 6, 3);
        $this->expectException(\InvalidArgumentException::class);
        $this->_testSubject->move(4, 3);
    }
    public function testPawn_Move_LegalCoordinates_Forward_UpdatesCoordinates () {
        $this->_chessBoard->add($this->_testSubject, 6, 3);
        $this->_testSubject->move(6, 2);
        $this->assertEquals(6, $this->_testSubject->getXCoordinate());
        $this->assertEquals(2, $this->_testSubject->getYCoordinate());
    }
}

File: ChessProject-Java/src/test/java/com/solarwindsmsp/chess/pieces/PawnTest.java
package com.solarwindsmsp.chess.pieces;

import com.solarwindsmsp.chess.*;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

public class PawnTest {

    private ChessBoard chessBoard;
    private Pawn testSubject;

    @Before
    public void setUp() {
        this.chessBoard = new ChessBoard();
        this.testSubject = new Pawn(PieceColor.BLACK);
    }

    @Test
    public void testChessBoard_Add_Sets_XCoordinate() throws IllegalBoardPlacementException {
        this.chessBoard.add(testSubject, 6, 3);
        assertEquals(6, testSubject.getXCoordinate());
    }

    @Test
    public void testChessBoard_Add_Sets_YCoordinate() throws IllegalBoardPlacementException {
        this.chessBoard.add(testSubject, 6, 3);
        assertEquals(3, testSubject.getYCoordinate());
    }


    @Test
    public void testPawn_Move_IllegalCoordinates_Right_DoesNotMove() throws IllegalBoardPlacementException {
        chessBoard.add(testSubject, 6, 3);
        try {
            testSubject.move(MovementType.MOVE, 7, 3);
        } catch (IllegalMoveException e) {
            e.printStackTrace();
        }
        assertEquals(6, testSubject.getXCoordinate());
        assertEquals(3, testSubject.getYCoordinate());
    }

    @Test
    public void testPawn_Move_IllegalCoordinates_Left_DoesNotMove() throws IllegalBoardPlacementException {
        chessBoard.add(testSubject, 6, 3);
        try {
            testSubject.move(MovementType.MOVE, 4, 3);
        } catch (IllegalMoveException e) {
            e.printStackTrace();
        }
        assertEquals(6, testSubject.getXCoordinate());
        assertEquals(3, testSubject.getYCoordinate());
    }

    @Test
    public void testPawn_Move_LegalCoordinates_Forward_UpdatesCoordinates() throws IllegalBoardPlacementException, IllegalMoveException {
        chessBoard.add(testSubject, 6, 3);
        testSubject.move(MovementType.MOVE, 6, 2);
        assertEquals(6, testSubject.getXCoordinate());
        assertEquals(2, testSubject.getYCoordinate());
    }

}